src/main/java/slimeknights/tconstruct/library/client/RenderUtil.java

```diff
@@ -173,7 +175,16 @@ public static void renderStackedFluidCuboid(FluidStack fluid, double px, double
     }
 
     int xd = to.getX() - from.getX();
-    int yd = (int) (ymax - ymin);
+
+    // the liquid can stretch over more blocks than the subtracted height is if ymin's decimal is bigger than ymax's decimal (causing UV over 1)
```

Comment:
Hence the correction in the quad below. The way it's done is to ensure correct calculation of UVs over block transitions. This will most likely only break things. Got a screenshot of what was wrong before?